### DISCUSSION TITLE
I want to reset the histogram metric data every 5 seconds.

### DISCUSSION THREAD

**Discussion by zerocooldog** [open-telemetry/opentelemetry-java-instrumentation#12795] [Q&A] [answered]
After reducing the metric collection interval to 5 seconds, I'm observing that histogram metrics are not resetting to zero at the beginning of the next collection cycle. 

Is this correct?

I want the metrics to start from zero every 5 seconds and measure data only for the next 5-second interval.

```
otel.metric.export.interval=5s

```

In conclusion, I want to reset the histogram metric data every 5 seconds.

![image](https://github.com/user-attachments/assets/f9d03b28-da82-41f0-8fba-cec1e2105b0e)

- When there is a request between 0 and 5 seconds:

```
       "histogram": {
          "dataPoints": [{
            "startTimeUnixNano": "1732593064682813600",
            "timeUnixNano": "1732595009705854200",
            "count": "5",
            "sum": 0.0524805,
            "bucketCounts": ["3", "1", "0", "1", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0"],
            "explicitBounds": [0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1.0, 2.5, 5.0, 7.5, 10.0],
            "attributes": [{
              "key": "http.request.method",
              "value": {
                "stringValue": "GET"
              }
            }, {
```

- When there is no request between 5 and 10 seconds:

```
       "histogram": {
          "dataPoints": [{
            "startTimeUnixNano": "1732593064682813600",
            "timeUnixNano": "1732595009705854200",
            "count": "5",
            "sum": 0.0524805,
            "bucketCounts": ["3", "1", "0", "1", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0"],
            "explicitBounds": [0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1.0, 2.5, 5.0, 7.5, 10.0],
            "attributes": [{
              "key": "http.request.method",
              "value": {
                "stringValue": "GET"
              }
            }, {
```

However, the data is not being reset to zero between 5 and 10 seconds.

**Comment by laurit** [comment]
> Is this correct?

Have a look at https://opentelemetry.io/docs/specs/otel/metrics/data-model/#histogram Values are rest for each reporting cycle only when delta aggregation temporality is used.

**Reply by zerocooldog** [reply]
@laurit I’m having trouble understanding "delta aggregation temporality." I’m not looking for a theoretical explanation, but rather how to configure it in the OTEL Java Agent.

The documentation explains that " The picture above denotes Delta temporality where accumulated event counts are reset to zero after reporting and a new aggregation occurs. Cumulative, on the other hand, continues to aggregate events, resetting with the use of a new start time."

In this context, how can I set the new start time?

The metrics log for http.server.request.duration in the question content is provided by OTEL, so I don't have a way to modify it.

**Reply by laurit** [reply, marked as answer]
> but rather how to configure it in the OTEL Java Agent.

See https://opentelemetry.io/docs/languages/java/configuration/#environment-variables-and-system-properties search for `otel.exporter.otlp.metrics.temporality.preference`

> The documentation explains that " The picture above denotes Delta temporality where accumulated event counts are reset to zero after reporting and a new aggregation occurs. Cumulative, on the other hand, continues to aggregate events, resetting with the use of a new start time."
> in this context, how can I set the new start time?

I don't know whether there is any other way besides restarting your application

**Reply by zerocooldog** [reply]
@laurit Thank you, it works as I wanted.
